src/odemis/driver/xt_client.py

```diff
@@ -1506,35 +1536,44 @@ def _acquire(self):
+                            self.parent.unblank_beam()
                     # The channel needs to be stopped to acquire an image, therefore immediately stop the channel.
-                    self.parent.set_channel_state(self._channel_name, False)
+                    self.parent.set_channel_state(self._scanner.channel, False)
+                    md = self._scanner._metadata.copy()
 
                     # Estimated time for an acquisition is the dwell time times the total amount of pixels in the image.
-                    n_pixels = self.parent._scanner.shape[0] * self.parent._scanner.shape[1]
-                    est_acq_time = self.parent._scanner.dwellTime.value * n_pixels
+                    if hasattr(self._scanner, "dwellTime") and hasattr(self._scanner, "shape"):
+                        n_pixels = self._scanner.resolution.value[0] * self._scanner.resolution.value[1]
+                        est_acq_time = self._scanner.dwellTime.value * n_pixels
+                        md[model.MD_DWELL_TIME] = self._scanner.dwellTime.value
+                    else:
+                        # Estimated acquisition time used the timeout if not all info about this acquisition time is
+                        # known.
+                        est_acq_time = 60
+
+                    if hasattr(self._scanner, "rotation"):
+                        md[model.MD_ROTATION] = self._scanner.rotation.value
 
                     # Wait for the acquisition to be received
                     logging.debug("Starting one image acquisition")
-                    try:
-                        if self._acq_wait_data(est_acq_time + 20):
-                            logging.debug("Stopping measurement early")
-                            self.stop_acquisition()
-                            break
-                    except TimeoutError as err:
-                        logging.error(err)
-                        self.stop_acquisition()
-                        break
-                    # Acquire the image
-                    image = self.parent.get_latest_image(self._channel_name)
-
-                    md = self.parent._scanner._metadata.copy()
+                    # try:
+                    #     if self._acq_wait_data(est_acq_time + 20):
+                    #         logging.debug("Stopping measurement early")
+                    #         self.stop_acquisition()
+                    #         break
+                    # except TimeoutError as err:
+                    #     logging.error(err)
+                    #     self.stop_acquisition()
+                    #     break
+
+                    # Retrieve the image
+                    image = self.parent.get_latest_image(self._scanner.channel)
                     md.update(self._metadata)
-                    md[model.MD_DWELL_TIME] = self.parent._scanner.dwellTime.value
-                    md[model.MD_ROTATION] = self.parent._scanner.rotation.value
```

Comment:
I tried getting the metadata after the image is pauzed but isn't yet taken (as early as possible but not to soon so that values might have changed. Plus there wasn't already a if statement for the dwelltime/rotation below.
Maybe it is to spread out though and it should be brought back down? I don't know.

Reply:
Ok, then just leave it as you have it now.